src/adapter/src/catalog.rs

```diff
@@ -4020,11 +4022,19 @@ impl<S: Append> Catalog<S> {
                             )));
                         }
                         if let ResolvedDatabaseSpecifier::Ambient = name.qualifiers.database_spec {
-                            let name = state
-                                .resolve_full_name(&name, session.map(|session| session.conn_id()));
-                            return Err(AdapterError::Catalog(Error::new(
-                                ErrorKind::ReadOnlySystemSchema(name.to_string()),
-                            )));
+                            // We allow users to create indexes on system objects to speed up
+                            // debugging related queries.
+                            if item.typ() != CatalogItemType::Index {
```

Comment:
It would be further possible to limit these to a certain subset of system objects. I figured that it would just add a bit of complexity and it won't hurt to allow people to create indexes on any type. Feel free to disagree.